test/EndToEnd/tests/ExecuteInitScriptTest.ps1

```diff
@@ -0,0 +1,150 @@
+function Test-NetCoreProjectExecuteInitScriptOnInstall {
+    param($context)
+
+    Remove-Variable PackageInitPS1Var -Scope Global -Force -ErrorAction Ignore
+
+    $componentModel = Get-VSComponentModel
+    $restoreEvents = $componentModel.GetService([NuGet.VisualStudio.IRestoreEvents])
+
+    $restoreEvent = $null
+
+    Get-Event | Remove-Event
+    Register-ObjectEvent -InputObject $restoreEvents -EventName SolutionRestoreCompleted -SourceIdentifier RestoreEventSource
+
+    Try {
+        $p = New-NetCoreConsoleApp
+
+        # Wait for initial restore
+        $restoreEvent = Wait-Event -SourceIdentifier RestoreEventSource -TimeoutSec 20
```

Comment:
A constant would be ideal, but my main point is to define this often repeated default at a higher scope (e.g.:  `DefaultRestoreTimeoutInSeconds`).